### ISSUE TITLE
Add random intercept values to the tidy method

### ISSUE THREAD

**Issue by seananderson** [pbs-assess/sdmTMB#41]
They are in the `sd_report`. If `m` is a fitted model:

```r
b <- as.list(m$sd_report, "Estimate")
b$RE
```

and you can reconstruct them from the `m$tmb_data$RE_indexes`. Where:

```c
RE(RE_indexes(i, k))
```

i is row of data, k is the random effect grouping ID, e.g. (1 | g ) + (1 | h) would have a k of 1 and 2.

**Comment by seananderson** [comment]
Awesome, thanks, can we add a unit test or two? Probably within this file https://github.com/pbs-assess/sdmTMB/blob/main/tests/testthat/test-random-intercepts.R

**Comment by ericward-noaa** [comment]
Yep, working on it. Tangential, is it worth including a ranef() or fixef() function like in glmmTMB? It'd be redundant with tidy()

**Comment by seananderson** [comment]
fixef() is already in there:
https://github.com/pbs-assess/sdmTMB/blob/7d2ae2c2f89c9ef4d5775a6338794ae61fc39a6d/R/methods.R#L69-L74

I think ranef() would be for things like omega_s, epsilon_st etc. so different

I fixed a typo in the new tidy() with `fit` left it.

Also I just realized the random intercept SDs (sigma_G) is already supposed to be in tidy() just not named.

See the current print() https://github.com/pbs-assess/sdmTMB/blob/449fadbe398679bfe2a689d0a27df3f0035bd7cf/R/print.R#L120-L132 which also has what may be a more succinct way of grabbing the names.

Oh, now I see what you did:

```
tidy(m, "ran_pars")
      term     estimate  std.error
1    range  1.048791530         NA
3      phi  0.100339857         NA
4  sigma_O  0.166866456         NA
5  sigma_G  0.434816305         NA
6  sigma_G  0.170371422         NA
8      g_1 -0.286447740 0.12401226
9      g_2  0.686187775 0.12378918
10     g_3  0.100281256 0.12411771
11     g_4 -0.314363744 0.12428128
12     g_5 -0.611679452 0.12405101
13     g_6  0.032951684 0.12402462
14     g_7  0.558095132 0.12388982
...
```

I think perhaps this should be another 'type'. One for parameters like sigma and another for the values themselves. Should look at what broom.mixed does.

**Comment by ericward-noaa** [comment]
I changed this a bit, adding a 'ranef' method to tidy() -- because broom.mixed::tidy() returns the random sigmas (not the random deviations when type = "ran_pars"). 

Also added ranef.sdmTMB() to methods.R -- with same behaviour as ranef.glmmTMB (minus the zero-inflated element)

And added tests https://github.com/pbs-assess/sdmTMB/blob/7dbf307fc56e793e7738e2a943cf50b85c2667d7/tests/testthat/test-random-intercepts.R#L116

**Comment by sschooler** [comment]
Is there a way to extract the names of the random effects (e.g. right now, they just return Sigma_G, and if you have multiple random effects, you don't know which one is which)?

**Comment by seananderson** [comment]
I think the part of tidy this issue relates does have names, but the part with 'ran_pars' does not. I'll post another issue with this question...
